code/game/objects/items/devices/radio/radio.dm

```diff
@@ -84,6 +94,13 @@
 	for(var/ch_name in channels)
 		secure_radio_connections[ch_name] = add_radio(src, GLOB.radiochannels[ch_name])
 
+	// allows you to just put an icon state for the var
```

Comment:
You shouldn't need this at all, our overlays system works perfectly fine for strings.

Reply:
the two _active overlays use flick_overlay_view instead of update_overlays

Reply:
Hmmm that is annoying

Instead of setting these to images on init (which limits what other code/admins can do, and also (less importantly) increases init times) you should instead create the images when they are actually needed, which is where making the flick a subproc of radios comes in handy. Ideally you would also cache it but that's not super important.